apps/ensadmin-next/src/app/ponder-client-api/provider.tsx

```diff
@@ -4,16 +4,15 @@ import { Provider as QueryClientProvider } from "@/components/query-client/provi
 import * as ponderSchema from "@ensnode/ponder-schema";
 import { createClient } from "@ponder/client";
 import { PonderProvider } from "@ponder/react";
+import { useState } from "react";
 
-import { selectedEnsNodeUrl } from "@/lib/env";
-import { useSearchParams } from "next/navigation";
-import { type PropsWithChildren, useState } from "react";
+type ProviderProps = {
+  url: string;
+  children: React.ReactNode;
+};
 
-export function Provider({ children }: PropsWithChildren) {
-  const searchParams = useSearchParams();
-  const [ponderClient] = useState(() =>
-    createPonderClient(selectedEnsNodeUrl(searchParams), schema),
-  );
+export function Provider({ url, children }: ProviderProps) {
```

Comment:
Passing URL as a prop, very nice improvement over having it calculated internally with search params.